File: Daybreak/Services/Scanner/Models/MapPayload.cs
namespace Daybreak.Services.Scanner.Models;

internal sealed class MapPayload
{
    public bool IsLoaded { get; set; }
    public uint Id { get; set; }
    public uint InstanceType { get; set; }
    public uint Timer { get; set; }
    public uint Campaign { get; set; }
    public uint Continent { get; set; }
    public uint Region { get; set; }
    public uint Type { get; set; }
}

File: Daybreak.GWCA/source/payloads/MapPayload.cpp
#include <cstdint>
#include <GWCA/GameEntities/Map.h>
#include <json.hpp>
#include <payloads/MapPayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const MapPayload& p) {
        j = json
        {
            {"IsLoaded", p.IsLoaded},
            {"Id", p.Id},
            {"InstanceType", p.InstanceType},
            {"Timer", p.Timer},
            {"Campaign", p.Campaign},
            {"Continent", p.Continent},
            {"Region", p.Region},
            {"Type", p.Type},
        };
    }
}